xbout/geometries.py

```diff
@@ -171,8 +171,13 @@ def add_toroidal_geometry_coords(ds, *, coordinates=None, grid=None):
     if 'z' in ds.dims:
         ds = ds.rename(z=coordinates['z'])
         nz = ds.dims[coordinates['z']]
-        phi = xr.DataArray(np.linspace(start=ds.metadata['ZMIN'],
-                                       stop=2 * np.pi * ds.metadata['ZMAX'], num=nz),
+        phi0 = 2*np.pi*ds.metadata['ZMIN']
+        phi1 = phi0 + nz*ds.metadata['dz']
+        if not np.isclose(phi1, 2.*np.pi*ds.metadata['ZMAX'], rtol=1.e-15, atol=0.):
+            warn('Size of toroidal domain as calculated from nz*dz (' + str(phi1 - phi0)
+                 + ' is not the same as 2pi*(ZMAX - ZMIN) ('
+                 + str(2.*np.pi*ds.metadata['ZMAX'] - phi0) + '): using value from dz')
```

Comment:
Possibly just personal preference, but I find using `'{}'.format(...)` more readable. Avoids the need to call `str`

Reply:
I use [f-strings](https://realpython.com/python-f-strings/) every time now: you don't need to call `str()` and you keep the argument in place in the text